File: design-system/android/components/DSColors.kt
package com.designsystem.theme

import androidx.compose.ui.graphics.Color

object DSColors {
    // Base Colors
    val White = Color(0xFFFFFFFF)
    val Black = Color(0xFF000000)
    
    // Gray Scale
    val Gray50 = Color(0xFFF9FAFB)
    val Gray100 = Color(0xFFF3F4F6)
    val Gray200 = Color(0xFFE5E7EB)
    val Gray300 = Color(0xFFD1D5DB)
    val Gray400 = Color(0xFF9CA3AF)
    val Gray500 = Color(0xFF6B7280)
    val Gray600 = Color(0xFF4B5563)
    val Gray700 = Color(0xFF374151)
    val Gray800 = Color(0xFF1F2937)
    val Gray900 = Color(0xFF111827)
    val Gray950 = Color(0xFF030712)
    
    // Primary Colors
    val Primary50 = Color(0xFFEFF6FF)
    val Primary100 = Color(0xFFDBEAFE)
    val Primary200 = Color(0xFFBFDBFE)
    val Primary300 = Color(0xFF93C5FD)
    val Primary400 = Color(0xFF60A5FA)
    val Primary500 = Color(0xFF3B82F6)
    val Primary600 = Color(0xFF2563EB)
    val Primary700 = Color(0xFF1D4ED8)
    val Primary800 = Color(0xFF1E40AF)
    val Primary900 = Color(0xFF1E3A8A)
    val Primary950 = Color(0xFF172554)
    
    // Secondary Colors
    val Secondary50 = Color(0xFFF0FDF4)
    val Secondary100 = Color(0xFFDCFCE7)
    val Secondary200 = Color(0xFFBBF7D0)
    val Secondary300 = Color(0xFF86EFAC)
    val Secondary400 = Color(0xFF4ADE80)
    val Secondary500 = Color(0xFF22C55E)
    val Secondary600 = Color(0xFF16A34A)
    val Secondary700 = Color(0xFF15803D)
    val Secondary800 = Color(0xFF166534)
    val Secondary900 = Color(0xFF14532D)
    val Secondary950 = Color(0xFF052E16)
    
    // Accent Colors
    val Accent50 = Color(0xFFFFF7ED)
    val Accent100 = Color(0xFFFFEDD5)
    val Accent200 = Color(0xFFFED7AA)
    val Accent300 = Color(0xFFFDBA74)
    val Accent400 = Color(0xFFFB923C)
    val Accent500 = Color(0xFFF97316)
    val Accent600 = Color(0xFFEA580C)
    val Accent700 = Color(0xFFC2410C)
    val Accent800 = Color(0xFF9A3412)
    val Accent900 = Color(0xFF7C2D12)
    val Accent950 = Color(0xFF431407)
    
    // Semantic Colors
    val ErrorLight = Color(0xFFFEE2E2)
    val ErrorMain = Color(0xFFEF4444)
    val ErrorDark = Color(0xFF991B1B)
    
    val WarningLight = Color(0xFFFEF3C7)
    val WarningMain = Color(0xFFF59E0B)
    val WarningDark = Color(0xFF92400E)
    
    val SuccessLight = Color(0xFFD1FAE5)
    val SuccessMain = Color(0xFF10B981)
    val SuccessDark = Color(0xFF065F46)
    
    val InfoLight = Color(0xFFDBEAFE)
    val InfoMain = Color(0xFF3B82F6)
    val InfoDark = Color(0xFF1E40AF)
    
    // Background Colors
    val BackgroundPrimary = White
    val BackgroundSecondary = Gray50
    val BackgroundTertiary = Gray100
    val BackgroundElevated = White
    val BackgroundOverlay = Black.copy(alpha = 0.5f)
    
    // Text Colors
    val TextPrimary = Gray900
    val TextSecondary = Gray600
    val TextTertiary = Gray400
    val TextDisabled = Gray300
    val TextInverse = White
    val TextLink = Primary600
    
    // Border Colors
    val BorderLight = Gray200
    val BorderDefault = Gray300
    val BorderDark = Gray400
    val BorderFocus = Primary500
    val BorderError = ErrorMain
}

File: design-system/ios/components/DSColors.swift
import SwiftUI

// MARK: - Design System Colors
public struct DSColors {
    // MARK: - Base Colors
    public static let white = Color(hex: "#FFFFFF")
    public static let black = Color(hex: "#000000")
    
    // MARK: - Gray Scale
    public static let gray50 = Color(hex: "#F9FAFB")
    public static let gray100 = Color(hex: "#F3F4F6")
    public static let gray200 = Color(hex: "#E5E7EB")
    public static let gray300 = Color(hex: "#D1D5DB")
    public static let gray400 = Color(hex: "#9CA3AF")
    public static let gray500 = Color(hex: "#6B7280")
    public static let gray600 = Color(hex: "#4B5563")
    public static let gray700 = Color(hex: "#374151")
    public static let gray800 = Color(hex: "#1F2937")
    public static let gray900 = Color(hex: "#111827")
    public static let gray950 = Color(hex: "#030712")
    
    // MARK: - Primary Colors
    public static let primary50 = Color(hex: "#EFF6FF")
    public static let primary100 = Color(hex: "#DBEAFE")
    public static let primary200 = Color(hex: "#BFDBFE")
    public static let primary300 = Color(hex: "#93C5FD")
    public static let primary400 = Color(hex: "#60A5FA")
    public static let primary500 = Color(hex: "#3B82F6")
    public static let primary600 = Color(hex: "#2563EB")
    public static let primary700 = Color(hex: "#1D4ED8")
    public static let primary800 = Color(hex: "#1E40AF")
    public static let primary900 = Color(hex: "#1E3A8A")
    public static let primary950 = Color(hex: "#172554")
    
    // MARK: - Secondary Colors
    public static let secondary50 = Color(hex: "#F0FDF4")
    public static let secondary100 = Color(hex: "#DCFCE7")
    public static let secondary200 = Color(hex: "#BBF7D0")
    public static let secondary300 = Color(hex: "#86EFAC")
    public static let secondary400 = Color(hex: "#4ADE80")
    public static let secondary500 = Color(hex: "#22C55E")
    public static let secondary600 = Color(hex: "#16A34A")
    public static let secondary700 = Color(hex: "#15803D")
    public static let secondary800 = Color(hex: "#166534")
    public static let secondary900 = Color(hex: "#14532D")
    public static let secondary950 = Color(hex: "#052E16")
    
    // MARK: - Accent Colors
    public static let accent50 = Color(hex: "#FFF7ED")
    public static let accent100 = Color(hex: "#FFEDD5")
    public static let accent200 = Color(hex: "#FED7AA")
    public static let accent300 = Color(hex: "#FDBA74")
    public static let accent400 = Color(hex: "#FB923C")
    public static let accent500 = Color(hex: "#F97316")
    public static let accent600 = Color(hex: "#EA580C")
    public static let accent700 = Color(hex: "#C2410C")
    public static let accent800 = Color(hex: "#9A3412")
    public static let accent900 = Color(hex: "#7C2D12")
    public static let accent950 = Color(hex: "#431407")
    
    // MARK: - Semantic Colors
    public static let errorLight = Color(hex: "#FEE2E2")
    public static let errorMain = Color(hex: "#EF4444")
    public static let errorDark = Color(hex: "#991B1B")
    
    public static let warningLight = Color(hex: "#FEF3C7")
    public static let warningMain = Color(hex: "#F59E0B")
    public static let warningDark = Color(hex: "#92400E")
    
    public static let successLight = Color(hex: "#D1FAE5")
    public static let successMain = Color(hex: "#10B981")
    public static let successDark = Color(hex: "#065F46")
    
    public static let infoLight = Color(hex: "#DBEAFE")
    public static let infoMain = Color(hex: "#3B82F6")
    public static let infoDark = Color(hex: "#1E40AF")
    
    // MARK: - Background Colors
    public static let backgroundPrimary = white
    public static let backgroundSecondary = gray50
    public static let backgroundTertiary = gray100
    public static let backgroundElevated = white
    public static let backgroundOverlay = Color.black.opacity(0.5)
    
    // MARK: - Text Colors
    public static let textPrimary = gray900
    public static let textSecondary = gray600
    public static let textTertiary = gray400
    public static let textDisabled = gray300
    public static let textInverse = white
    public static let textLink = primary600
    
    // MARK: - Border Colors
    public static let borderLight = gray200
    public static let borderDefault = gray300
    public static let borderDark = gray400
    public static let borderFocus = primary500
    public static let borderError = errorMain
}

// MARK: - Color Extension for Hex Support
extension Color {
    init(hex: String) {
        let hex = hex.trimmingCharacters(in: CharacterSet.alphanumerics.inverted)
        var int: UInt64 = 0
        Scanner(string: hex).scanHexInt64(&int)
        let a, r, g, b: UInt64
        switch hex.count {
        case 3: // RGB (12-bit)
            (a, r, g, b) = (255, (int >> 8) * 17, (int >> 4 & 0xF) * 17, (int & 0xF) * 17)
        case 6: // RGB (24-bit)
            (a, r, g, b) = (255, int >> 16, int >> 8 & 0xFF, int & 0xFF)
        case 8: // ARGB (32-bit)
            (a, r, g, b) = (int >> 24, int >> 16 & 0xFF, int >> 8 & 0xFF, int & 0xFF)
        default:
            (a, r, g, b) = (255, 0, 0, 0)
        }
        
        self.init(
            .sRGB,
            red: Double(r) / 255,
            green: Double(g) / 255,
            blue: Double(b) / 255,
            opacity: Double(a) / 255
        )
    }
}